mono/mini/abcremoval.h

```diff
@@ -179,19 +179,28 @@ typedef struct MonoRelationsEvaluationRanges {
 
 /**
  * The context of a variable evaluation.
- * status: the evaluation status
  * current_relation: the relation that is currently evaluated.
  * ranges: the result of the evaluation.
  * father: the context of the evaluation that invoked this one (used to
  *         perform the backtracking when loops are detected.
  */
 typedef struct MonoRelationsEvaluationContext {
-	MonoRelationsEvaluationStatus status;
 	MonoSummarizedValueRelation *current_relation;
 	MonoRelationsEvaluationRanges ranges;
 	struct MonoRelationsEvaluationContext *father;
 } MonoRelationsEvaluationContext;
 
+/**
+ * This is a performance optimization. Clean_context was
+ * coming to dominate the running time of abcremoval. By
+ * storing the statuses together, we can memset the entire
+ * region.
+ */ 
+typedef struct MonoRelationsEvaluationContextContainer {
+	MonoRelationsEvaluationStatus *statuses;
+	MonoRelationsEvaluationContext *contexts;
+} MonoRelationsEvaluationContextContainer;
+
 /*
```

Comment:
As I'm embedding this struct directly rather than having a pointer to it, this wouldn't be any indirection at all. I was choosing to keep the two arrays in the same struct because they're parallel arrays; a given index into both arrays should refer to the same data. I can place both in the other struct instead if that's the style that's preferred.